core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,284 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/**
+ * @file PerfTest_Stream.cpp
+ * @brief Implementation of STREAM benchmark operations for Kokkos.
+ *
+ * @details This file provides a set of memory bandwidth benchmarks based on the
+ * STREAM benchmark suite. It implements the five core STREAM operations (Set,
+ * Copy, Scale, Add, and Triad) using Kokkos parallel primitives. It includes
+ * validation.
+ *
+ * The implementation strives to use as few Kokkos features as possible, thus
+ * validation is performed on the host rather than via parallel_reduce.
+ */
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+constexpr static double A_INIT = 1.0;
+constexpr static double B_INIT = 2.0;
+constexpr static double C_INIT = 3.0;
+constexpr static double SCALAR = 4.0;
+
+template <unsigned MemTraits>
+using StreamView = Kokkos::View<double*, Kokkos::MemoryTraits<MemTraits>>;
```

Comment:
STREAM is not templated on the data type, it's always double precision fps?

Reply:
The standard STREAM is in terms of doubles. We could parameterize.